cql3/expr/prepare_expr.cc

```diff
@@ -1483,7 +1483,7 @@ ::shared_ptr<assignment_testable> as_assignment_testable(expression e, std::opti
 static const column_value resolve_column(const unresolved_identifier& col_ident, const schema& schema) {
     ::shared_ptr<column_identifier> id = col_ident.ident->prepare_column_identifier(schema);
     const column_definition* def = get_column_definition(schema, *id);
-    if (!def || def->is_hidden_from_cql()) {
+    if (!def || def->is_view_virtual()) {
```

Comment:
I want to add a note about the commit message:

It's not just that view_virtual columns are undocumented and have a weird type - the biggest problem is that they have exactly the same name as the columns in the base table.
In other words, if a user that accidentally SELECTs from a view a column which exists in the base table but wasn't SELECTed into the view, the user will get this "empty" value instead of the error that it is. This is a very common mistake to make and we need good error checking for it, not strange silent responses.